dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -189,31 +314,60 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
           Future.successful(value)
         case None =>
           Future.failed(
-            new RuntimeException(
+            new IllegalArgumentException(
               s"No event saved with nonce ${nonce.hex} $outcome"))
       }
 
-      eventOutcomeOpt <- eventOutcomeDAO.read((nonce, outcome))
+      eventOutcomeOpt <- eventOutcomeDAO.read((nonce, outcome.outcomeString))
       eventOutcomeDb <- eventOutcomeOpt match {
         case Some(value) => Future.successful(value)
         case None =>
-          Future.failed(new RuntimeException(
-            s"No event outcome saved with nonce and message ${nonce.hex} $outcome"))
+          Future.failed(new IllegalArgumentException(
+            s"No event outcome saved with nonce and message ${nonce.hex} ${outcome.outcomeString}"))
       }
 
-      sig = eventDb.signingVersion match {
-        case Mock =>
-          val kVal = getKValue(rValDb, Mock)
-          require(kVal.schnorrNonce == rValDb.nonce,
+      sigVersion = eventDb.signingVersion
+
+      kVal = getKValue(rValDb, sigVersion)
+      _ = require(kVal.schnorrNonce == rValDb.nonce,
                   "The nonce from derived seed did not match database")
-          signingKey.schnorrSignWithNonce(eventOutcomeDb.hashedMessage.bytes,
-                                          kVal)
-      }
+
+      hashBytes = eventOutcomeDb.hashedMessage
+      sig = signingKey.schnorrSignWithNonce(hashBytes, kVal)
 
       updated = eventDb.copy(attestationOpt = Some(sig.sig))
       _ <- eventDAO.update(updated)
     } yield updated
   }
+
+  def signLargeRange(
+      eventDescriptorTLV: LargeRangeEventDescriptorV0TLV,
+      num: Long): Future[OracleEvent] = {
+
+    // Make this a vec so it is easier to add on
+    val signSigF =
+      eventDescriptorTLV.signedNonceOpt match {
+        case Some(nonce) =>
+          val signOutcome = LargeRangeSignAttestation(num >= 0)
+          signEvent(nonce, signOutcome).map(db => Vector(db))
+        case None =>
+          FutureUtil.emptyVec[EventDb]
+      }
+
+    val decomposed = NumberUtil.decompose(Math.abs(num),
+                                          eventDescriptorTLV.base.toInt,
+                                          eventDescriptorTLV.digitNonces.size)
+    val digitSigFs = eventDescriptorTLV.digitNonces.reverse.zipWithIndex.map {
```

Comment:
why does this need to be reversed? Does signing order matter here?

Reply:
Yes

https://github.com/discreetlogcontracts/dlcspecs/pull/55#discussion_r507920129